src/core/geometry/qgsgeometry.cpp

```diff
@@ -266,6 +266,7 @@ QgsGeometry QgsGeometry::collectGeometry( const QVector< QgsGeometry > &geometri
       }
       else
       {
+        collected.convertToMultiType();
```

Comment:
The logic here looks all off -- I can't see how g.isMultipart would ever be false at line 260, since it's gauranteed to be multipart at line 256. Which would mean that this condition is impossible to reach. Or am i missing something?

Reply:
I think you got confused. `Collected` is the output geometry, guaranteed to be multipart in 256.
`g` is iteratively each of the parameter geometries and it simply may or may not be multipart, so the code is not unreachable. The addition of line 269 is however redundant  since collected is guaranteed to be multipart by line 256.